pkg/config/store.go

```diff
@@ -78,6 +78,10 @@ type ChangeNotifier interface {
 	// The client should issue ReadChangeLog to see what has changed if the call is available.
 	// else it should re-read the store, perform diff and apply changes.
 	RegisterStoreChangeListener(s StoreListener)
+
+	// IsStoreChangeAvailable returns true only when the backend store is configured
+	// to emit change notifications.
+	IsStoreChangeAvailable() bool
```

Comment:
I'm not convinced we need this. Why does it matter if my backend emits change notifications? This is a `ChangeNotifier`, the thing implementing this interface is responsible for producing change notifications, the fact that the backend this is abstracting over may or may not natively support change notifications seems like a detail that doesn't need to be exposed - my implementation of this interface is responsible for synthesizing notifications as appropriate.

Plus, as far as I can tell it's only used in a test.